### DISCUSSION TITLE
Compiling on Apple Silicon

### DISCUSSION THREAD

**Discussion by tjbcg** [ahrm/sioyek#1031] [Q&A]
Hi there ! I'm trying to compile sioyek's dev branch on my mac but I can't get it to work.

I've installed dependencies: `brew install freeglut mesa harfbuzz qt`, I've added qmake to my `$PATH`: `set -x PATH /opt/homebrew/opt/qt@6/bin $PATH`.

Running the `build_mac.sh` does indeed compile mupdf and sioyek succesfully but linking fails. [Here](https://pastebin.com/raw/Dpgm12qS) is the full output.

I'm on Sonoma, on an M1 macbook pro. Also Xcode is installed (latest version afaik). What am I missing ?

**Comment by ahrm** [comment]
What is the version of the installed `qt` package? Also are you building the `main` branch or the `development` branch?

**Reply by tjbcg** [reply]
I'm building the dev branch, and qt is 6.6.2.
I did get past the linking issue by changing the MACOS_TARGET to 14 instead of 11, but now I have some other error (Cannot resolve rpath...).
I found some stuff regarding this in the issues, I'll try it out and report back.

**Reply by tjbcg** [reply]
I got it working, turns out homebrew's qt has a bunch of issues with macdeployqt. For future reference, [this](https://gist.github.com/FliiFe/9c058bee1b7cfedad6c724fd3e337851) is what worked for me, which can be helpful to fix the macOS github actions (they install qt5 instead of qt6).

In passing, you can probably do away with qt5 entirely, since the qt6 macros introduced some syntax error with qt5 (off the top of my head, `num_repeats_f_y` is defined only with qt6 but used with qt5 in main_widget.cpp in current development HEAD).

**Reply by 5183nischal** [reply]
Hey @FliiFe, I tried your formula for building on Apple Silicon, and I got error saying:
`MAKE_PARALLEL set to 8
    LINK build/release/mutool
ld: warning: -s is obsolete
ld: library 'crypto' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [build/release/mutool] Error 1
`
Do you have any suggestion/ideas how I can fix it and build? Thank you!

**Reply by tjbcg** [reply]
@5183nischal have you tried installing `openssl` via homebrew ? If that fixes it, I'll add it to the script.

**Reply by 5183nischal** [reply]
@FliiFe I had `openssl` from brew but it still gave an error but when I reinstalled with `brew reinstall openssl`, it all worked! Thank you!
